main.tf

```
@@ -21,7 +21,8 @@ locals {
     autoShutdown = var.autoShutdown
   }
 
-  all_tags = merge(local.common_tags, local.additional_tags)
+  expiresAfter = var.expiresAfter == "0000-00-00" ? formatdate("YYYY-MM-DD", timeadd(timestamp(), "720h")) : var.expiresAfter
+  all_tags = local.common_tags.environment == "sandbox" ? merge(local.common_tags, local.additional_tags, tomap({"expiresAfter" = local.expiresAfter})) : merge(local.common_tags, local.additional_tags)
```

this would be simpler if split out I think, something like this (untested):

```suggestion
  expires_tags = local.common_tags.environment == "sandbox" ? tomap({"expiresAfter" = local.expiresAfter})) : {}
  all_tags = merge(local.common_tags, local.additional_tags, local.expires_tags)
```

---

why revert? you will need to test the syntax slightly may need a `tomap({})` instead of `{}`